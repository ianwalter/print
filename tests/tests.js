const { test } = require('@ianwalter/bff')
const execa = require('execa')
const stripAnsi = require('strip-ansi')
const { createPrint } = require('..')

test('print', async t => {
  const env = { DEBUG: 'app.*', FORCE_COLOR: '2' }
  const { stdout } = await execa('pnpm', ['example', '-s'], { env })
  stdout.split('\n').forEach(line => {
    // Don't assert stacktrace lines.
    if (!stripAnsi(line).match(/ {4}at /)) {
      t.expect(line).toMatchSnapshot()
    }
  })
})

test('return', t => {
  const print = createPrint({ io: false })
  t.expect(print.info('Ello Guvna')).toMatchSnapshot()
})
