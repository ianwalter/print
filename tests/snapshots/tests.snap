// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print 1`] = `
"$ node tests/fixtures/example.js
🐛  Flaky test started. 
    Make sure you check it out.
    Could be trouble.
💬  Request made to server.
🔑  $2b$12$HMJFAblrhBCGxTWv5BnIFe
    export default () => {
      console.log('Hello World!')
    }

✅  You did it! Great job."
`;
